lib/metadata/bucketclient/backend.js

```diff
@@ -58,15 +58,15 @@ class BucketClientInterface {
         return null;
     }
 
-    getBucketAndObject(bucketName, objName, log, cb, params) {
+    getBucketAndObject(bucketName, objName, params, log, cb) {
         const _params = this.extendedVersion ? params : undefined;
```

Comment:
Now that query is fully supported in metadata, the `this.extendedVersion` can be removed.